4_gap_analysis/src/1b_phys_byState.R

```diff
@@ -0,0 +1,130 @@
+}else{
+  state<-'New_Mexico'
+}
+
+# Cons for this script - pH, SC, Temp
+Con<-'temperature'
+StartDate<-'1986-01-01'
+EndDate<-'2015-12-31'
+
+# Bad values to remove
+if (Con == 'ph') badUnits=c('%')
+if (Con == 'SC') badUnits=c('%','m','mg/l','None','',' ')
+if (Con == 'temperature') badUnits=c('%','m','mg/l','None','',' ','umho/cm','mS/cm','mg/l      ','umho/cm   ')
+
+# Crosswalk based on Agency name, NDA acroym, and WQX agency code
+xWalk2<-loadWorkbook('4_gap_analysis/in/Organizations_WQP.xlsx') %>% readWorksheet(sheet='WQP_Organizations_mlr',header=T)
+
+# Designate state abbreviation
+state2<-gsub('_',' ',state)
+stateAb<-state.abb[grep(state2, state.name)][1]
+# State abbreviations for states not covered by R state abbreviation database
+if(state == 'Northern_Mariana_Islands') stateAb='MP'
+if(state == 'Guam') stateAb='GU'
+if(state == 'American_Samoa') stateAb='AS'
+if(state == 'District_of_Columbia') stateAb='DC'
+if(state == 'Puerto_Rico') stateAb='PR'
+if(state == 'Virgin_Islands') stateAb='VI'
+
+# Reading files locally
+all_files <- dir('1_wqpdata/out/byState', pattern=paste0("^",state),full.names=T) %>% #pattern=paste0('*',state,'*'),full.names = T) %>%
+  grep(paste('(',Con,')',sep=''), ., value=TRUE)
+data_files<-all_files[!grepl('siteinfo|ind',all_files)] # just the data files
+site_files<-all_files[grepl('siteinfo|ind',all_files)] # just the site files
+
+# combine into one data frame
+combined_data <- bind_rows(
+  lapply(data_files, function(dfile) {
+    # By State
+    dat <- feather::read_feather(dfile) %>%
+      # Below here are data coercion fixes for columns
+      mutate_at(vars(starts_with('ResultDepthHeightMeasure.MeasureValue'),starts_with('ActivityTopDepthHeightMeasure.MeasureValue'),
+                     starts_with('ActivityBottomDepthHeightMeasure.MeasureValue'),starts_with('ResultAnalyticalMethod.MethodIdentifier'),
+                     starts_with('ProjectIdentifier'),starts_with('SampleCollectionMethod.MethodIdentifier'),
+                     starts_with('ActivityCommentText'),starts_with('LaboratoryName')),
+                list(as.character)) %>%
+      mutate(ConstituentGroup=Con) %>%
+      # filter out quality control rows and media types
+      filter(!grepl('Quality Control',ActivityTypeCode) & ActivityMediaName %in% c('Water','water','Sediment') &
+              ActivityMediaSubdivisionName %in% c('Surface Water',NA,'','Unknown') &
+               !ResultMeasure.MeasureUnitCode %in%  badUnits &
+               !is.na(ResultMeasure.MeasureUnitCode) &
+              ActivityStartDate > as.Date(StartDate) & ActivityStartDate < as.Date(EndDate))
+  })
+) %>% # Activity Start Date is not completed, can't be confident when measurement was actually taken
+  filter(!is.na(ActivityStartDate))
+
+# Both these fields may have valid measurement or concentration values
+# This keeps rows that have data in or the other
+combined_data<-subset(combined_data,(!is.na(combined_data$ResultMeasureValue)) |
```

Comment:
Could add to pipe chain above and use `filter` instead, as described in `1a_DO_byState.R`